qiita_db/analysis.py

```diff
@@ -165,6 +169,71 @@ def create(cls, owner, name, description, parent=None, from_default=False):
         a_id = conn_handler.execute_queue(queue)[0]
         return cls(a_id)
 
+    @classmethod
+    def delete(cls, _id):
```

Comment:
@antgonza there is still another table in which the analysis can be present: `analysis_chain`. Right now we are not using that table for anything, but we can easily forget about it on deletion once we start using it. There are 2 possible solutions:
 * Open an issue, so we don't keep track of it
 * We add that table here. This implies that you also need to add a check: a study cannot be deleted if it has 'child' analyses.

I'm fine with adding an issue, since currently we are not using that and a lot of other changes will be needed once that gets in place.